### ISSUE TITLE
running out of space on batch instances

### ISSUE THREAD

**Issue by chrisgorgo** [OpenNeuroOrg/openneuro#487]
```
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/anat/sub-1102_T1w.json to ../snapshot/data/sub-1102/anat/sub-1102_T1w.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/anat/sub-1102_T2w.json to ../snapshot/data/sub-1102/anat/sub-1102_T2w.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/dwi/sub-1102_dwi.bval to ../snapshot/data/sub-1102/dwi/sub-1102_dwi.bval [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/dwi/sub-1102_dwi.json to ../snapshot/data/sub-1102/dwi/sub-1102_dwi.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/dwi/sub-1102_dwi.bvec to ../snapshot/data/sub-1102/dwi/sub-1102_dwi.bvec [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/anat/sub-1102_T2w.nii.gz to ../snapshot/data/sub-1102/anat/sub-1102_T2w.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/fmap/sub-1102_magnitude2.nii.gz to ../snapshot/data/sub-1102/fmap/sub-1102_magnitude2.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/fmap/sub-1102_magnitude1.nii.gz to ../snapshot/data/sub-1102/fmap/sub-1102_magnitude1.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/fmap/sub-1102_phasediff.json to ../snapshot/data/sub-1102/fmap/sub-1102_phasediff.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-2step_events.tsv to ../snapshot/data/sub-1102/func/sub-1102_task-2step_events.tsv [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/fmap/sub-1102_phasediff.nii.gz to ../snapshot/data/sub-1102/fmap/sub-1102_phasediff.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-2step_run-01_bold.json to ../snapshot/data/sub-1102/func/sub-1102_task-2step_run-01_bold.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-2step_run-02_bold.json to ../snapshot/data/sub-1102/func/sub-1102_task-2step_run-02_bold.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-2step_run-03_bold.json to ../snapshot/data/sub-1102/func/sub-1102_task-2step_run-03_bold.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1101/func/sub-1101_task-2step_run-04_bold.nii.gz to ../snapshot/data/sub-1101/func/sub-1101_task-2step_run-04_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1101/func/sub-1101_task-2step_run-03_bold.nii.gz to ../snapshot/data/sub-1101/func/sub-1101_task-2step_run-03_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1101/func/sub-1101_task-2step_run-05_bold.nii.gz to ../snapshot/data/sub-1101/func/sub-1101_task-2step_run-05_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1101/func/sub-1101_task-2step_run-06_bold.nii.gz to ../snapshot/data/sub-1101/func/sub-1101_task-2step_run-06_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1101/func/sub-1101_task-rest_run-01_bold.nii.gz to ../snapshot/data/sub-1101/func/sub-1101_task-rest_run-01_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-2step_run-06_bold.json to ../snapshot/data/sub-1102/func/sub-1102_task-2step_run-06_bold.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-rest_run-02_bold.json to ../snapshot/data/sub-1102/func/sub-1102_task-rest_run-02_bold.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-2step_run-01_bold.nii.gz to ../snapshot/data/sub-1102/func/sub-1102_task-2step_run-01_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-2step_run-02_bold.nii.gz to ../snapshot/data/sub-1102/func/sub-1102_task-2step_run-02_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-2step_run-03_bold.nii.gz to ../snapshot/data/sub-1102/func/sub-1102_task-2step_run-03_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-2step_run-04_bold.nii.gz to ../snapshot/data/sub-1102/func/sub-1102_task-2step_run-04_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-2step_run-05_bold.nii.gz to ../snapshot/data/sub-1102/func/sub-1102_task-2step_run-05_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-rest_run-01_bold.nii.gz to ../snapshot/data/sub-1102/func/sub-1102_task-rest_run-01_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-rest_run-02_bold.nii.gz to ../snapshot/data/sub-1102/func/sub-1102_task-rest_run-02_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1102/func/sub-1102_task-2step_run-06_bold.nii.gz to ../snapshot/data/sub-1102/func/sub-1102_task-2step_run-06_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1104/func/sub-1104_task-2step_run-01_bold.nii.gz to ../snapshot/data/sub-1104/func/sub-1104_task-2step_run-01_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1104/func/sub-1104_task-2step_run-02_bold.nii.gz to ../snapshot/data/sub-1104/func/sub-1104_task-2step_run-02_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1104/func/sub-1104_task-rest_run-02_bold.json to ../snapshot/data/sub-1104/func/sub-1104_task-rest_run-02_bold.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/task-rest_bold.json to ../snapshot/data/task-rest_bold.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/task-2step_bold.json to ../snapshot/data/task-2step_bold.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/task-2step_events.json to ../snapshot/data/task-2step_events.json [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1104/func/sub-1104_task-2step_run-04_bold.nii.gz to ../snapshot/data/sub-1104/func/sub-1104_task-2step_run-04_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1104/func/sub-1104_task-2step_run-03_bold.nii.gz to ../snapshot/data/sub-1104/func/sub-1104_task-2step_run-03_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1104/func/sub-1104_task-2step_run-05_bold.nii.gz to ../snapshot/data/sub-1104/func/sub-1104_task-2step_run-05_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1104/func/sub-1104_task-rest_run-01_bold.nii.gz to ../snapshot/data/sub-1104/func/sub-1104_task-rest_run-01_bold.nii.gz [Errno 28] No space left on device
download failed: s3://openneuro.snapshots/b5ea7c3ba01c27b183312a9c2f017d9d/sub-1104/func/sub-1104_task-2step_run-06_bold.nii.gz to ../snapshot/data/sub-1104/func/sub-1104_task-2step_run-06_bold.nii.gz [Errno 28] No space left on device
```
https://openneuro.org/datasets/ds001271/versions/00001?app=MRIQC&version=65&job=5aa969e7ba9c9a0001dc56ad

This is surprising because data for this subtask should only required 6-7Gb (two subjects)

**Comment by nellh** [comment]
The group level does copy the whole dataset which makes sense it would run out of space with the 175GB volume but that does not explain why the participant levels failed.

**Comment by chrisgorgo** [comment]
Agreed. The weird thing is that all of the participant subjobs failed.

**Comment by chrisgorgo** [comment]
I could not check logs due to #490 but this job also has empty results which could be due to space issue : https://openneuro.org/datasets/ds001269/versions/00003?app=FMRIPREP&version=54&job=5aaa479b02c03f0001e82d07

**Comment by chrisgorgo** [comment]
This happens even if analysis of a single subject is scheduled (https://openneuro.org/datasets/ds001271/versions/00001/file-display/MRIQC-ICA%2Fdefault%2F6349f2a5-7bf5-4220-a2e7-1127b932e822) - I think there might be something wrong with the s3 sync procedure.

**Comment by nellh** [comment]
Can I rerun one of these with one participant on production? Running the job locally with the bids-app-host container it correctly copies only the participant files but there could be an escaping issue somewhere between OpenNeuro and Batch causing this. It would speed up debugging to have a running job to inspect.

**Comment by chrisgorgo** [comment]
Go ahead.

Best,
Chris

**Comment by chrisgorgo** [comment]
One way to make the system more robust to those situation is to use EFS for shared scratch space.

**Comment by chrisgorgo** [comment]
This is still an issue: https://openneuro.org/datasets/ds001271/versions/00001/file-display/MRIQC%2Fdefault%2Fe8a4df62-357e-4f8c-8ce3-8408ec6b8593

**Comment by nellh** [comment]
@chrisfilo That's my fault, I messed up creating the tag on Dockerhub, it is fixed now but may be up to 3 hours to be live.

`0.8.4` at the top of the log is a fixed job, `0.8.3` has this bug still.
